paparazzi/src/main/java/app/cash/paparazzi/internal/resources/PseudolocaleGenerator.kt

```diff
@@ -0,0 +1,144 @@
+package app.cash.paparazzi.internal.resources
+
+import app.cash.paparazzi.internal.resources.base.BasicArrayResourceItem
+import app.cash.paparazzi.internal.resources.base.BasicPluralsResourceItem
+import app.cash.paparazzi.internal.resources.base.BasicTextValueResourceItem
+import app.cash.paparazzi.internal.resources.base.BasicValueResourceItem
+import app.cash.paparazzi.internal.resources.base.BasicValueResourceItemBase
+import com.android.ide.common.rendering.api.ResourceNamespace
+import com.android.ide.common.resources.ResourceValueMap
+import com.android.ide.common.resources.configuration.FolderConfiguration
+import com.android.ide.common.resources.configuration.LocaleQualifier
+import com.android.resources.ResourceType
+import com.google.common.collect.Table
+
+private const val DO_NOT_TRANSLATE = "donottranslate"
+
+private val configCache = hashMapOf<Pair<ResourceSourceFile, LocaleQualifier>, RepositoryConfiguration>()
+
+fun String.pseudoLocalize(method: Pseudolocalizer.Method) =
+  if (isBlank()) {
+    this
+  } else {
+    with(Pseudolocalizer(method)) {
+      start() + text(this@pseudoLocalize) + end()
+    }
+  }
+
+fun Table<ResourceNamespace, ResourceType, ResourceValueMap>.pseudolocalizeIfNeeded(
+  localeQualifier: LocaleQualifier
+): Table<ResourceNamespace, ResourceType, ResourceValueMap> {
+  for (namespace in rowKeySet()) {
+    for (type in columnKeySet()) {
+      val map = this[namespace, type]
+      map?.forEach { (name, value) ->
+        map[name] =
+          (value as? BasicValueResourceItemBase)?.pseudolocalizeIfNeeded(localeQualifier) ?: value
+      }
+    }
+  }
+
+  return this
+}
+
+private fun BasicValueResourceItemBase.pseudolocalizeIfNeeded(
+  localeQualifier: LocaleQualifier
+): BasicValueResourceItemBase {
+  if (!isPseudolocalizable(this)) {
```

Comment:
suggestion: pull this up into the caller and make a private extension for better readability

```
        map.forEach { (key, value) ->
          if(value !is BasicValueResourceItemBase || !value.isPseudolocalizable()) {
            return@forEach
          }
```